src/graphql/entries.ts

```diff
@@ -1,39 +1,96 @@
 import { ContentTypeProps, EntryProps } from 'contentful-management/types';
 
+import { CollectionItem, SysProps } from '../types';
 import { isPrimitiveField, logUnrecognizedFields } from './utils';
 
 /**
  * Updates GraphQL response data based on CMA entry object
  *
- * @param contentType entity
- * @param data the GraphQL response to be updated
- * @param update CMA entry object containing the update
- * @param locale code
+ * @param contentType ContentTypeProps
+ * @param data Record<string, unknown> - The GraphQL response to be updated
+ * @param update EntryProps - CMA entry object containing the update
+ * @param locale string - Locale code
+ * @returns Record<string, unknown> - Updated GraphQL response data
  */
 export function updateEntry(
   contentType: ContentTypeProps,
-  data: Record<string, unknown>,
+  data: Record<string, unknown> & { sys: SysProps },
   update: EntryProps,
   locale: string
-): Record<string, unknown> {
+): Record<string, unknown> & { sys: SysProps } {
   const modified = { ...data };
   const { fields } = contentType;
 
-  // Warn about unrecognized fields
   logUnrecognizedFields(
```

Comment:
the log and the clone of data into modified could be after the check if the sys.id matches ;)